src/Data/List/Linear.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Data.List.Linear
+  where
+
+import qualified Unsafe.Linear as Unsafe
+import qualified Prelude as Prelude
+import Prelude (Bool, Maybe(..), Either(..), Int)
+import Data.Num.Linear
+import Prelude.Linear.Internal.Simple
+import Data.Unrestricted.Linear
+import Data.Functor.Linear
+import Data.List.NonEmpty (NonEmpty ((:|)))
+
+map :: (a #-> b) -> [a] #-> [b]
+map = fmap
+
+-- | @filter p xs@ returns a list with elements satisfying the predicate.
+--
+-- See 'Data.Maybe.Linear.mapMaybe' if you do not want the 'Dupable' constraint.
+filter :: Dupable a => (a #-> Bool) -> [a] #-> [a]
```

Comment:
I think it's a good idea to have one with a consumable constraint instead -- using `a #-> (Bool, a)`. Some things might be easy to consume but hard to duplicate, like mutable arrays.

Reply:
You are right, that is actually the reason I mentioned `Data.Maybe.Linear.mapMaybe` on the Haddock. Its type signature(`(a #-> Maybe b) -> [a] #-> [b]`) is similar to yours, but I think it's a bit nicer because with returning `Maybe a` we don't need to get `(False, a)` just to immediately forget the `a`. But if you think it's still necessary I am happy to add that function too.

Reply:
Actually nevermind. Perhaps just mentioning that for consumable things, you can easily use `mapMaybe` in the haddock would be sufficient.